File: my-folder/0027-remove-element/solution.java
class Solution {
    public int removeElement(int[] nums, int val) {
        if(nums.length == 0){
            return 0;
        }
        int index = 0;
        for(int i=0; i<nums.length; i++){
            // if nums[i] is not val, insert into index
            if(nums[i] != val){
                nums[index ++] = nums[i];
            }
            // else if nums[i] is val skip
        }

        return index;
    }
}

File: my-folder/0027-remove-element/solution.py
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        p = 0
        for i in range(len(nums)):
            if val != nums[i]:
                nums[p] = nums[i]
                p += 1 
        
        return p 
